dataplicity/client.py

```diff
@@ -95,31 +97,66 @@ def disk_poll(self):
 
         if now >= self.next_disk_poll_time:
             self.next_disk_poll_time = now + self.disk_poll_rate_seconds
-            disk_space = disk_usage('/')
+            disk_space = disk_usage("/")
 
             with self.remote.batch() as batch:
                 batch.call_with_id(
-                    'authenticate_result',
-                    'device.check_auth',
-                    device_class='tuxtunnel',
+                    "authenticate_result",
+                    "device.check_auth",
+                    device_class="tuxtunnel",
                     serial=self.serial,
-                    auth_token=self.auth_token
+                    auth_token=self.auth_token,
                 )
                 batch.call_with_id(
-                    'set_disk_space_result',
-                    'device.set_disk_space',
+                    "set_disk_space_result",
+                    "device.set_disk_space",
                     disk_capacity=disk_space.total,
-                    disk_used=disk_space.used
+                    disk_used=disk_space.used,
                 )
 
+    def tag_poll(self):
+        """Gets the tag list for get_tag_list() and sends to the server"""
+        try:
+            tag_list = get_tag_list()
+        except TagException:
+            return
+
+        if tag_list != self._tag_list:
+            with self.remote.batch() as batch:
+                batch.call_with_id(
+                    "authenticate_result",
+                    "device.check_auth",
+                    device_class="tuxtunnel",
+                    serial=self.serial,
+                    auth_token=self.auth_token,
+                )
+                batch.call_with_id(
+                    "set_machine_defined_tags_result",
+                    "device.set_machine_defined_tags",
+                    tag_list=tag_list,
+                )
+            try:
+                batch.check("set_machine_defined_tags_result")
+            except Exception as e:
```

Comment:
Best to avoid pokemon exceptions when we can predict what kind of exceptions will be thrown. In this case I believe it is a JSONRPCError, but please check `jsonrpc.py` (it won't bite).